### ISSUE TITLE
BlackArch + Existing Arch ---> MUST READ!!! - FIXES ALL EXISTING ARCH INSTALL ERRORS

### ISSUE THREAD

**Issue by CommanderTurtle** [BlackArch/blackarch#967]
Mods, admins, please make this a sticky or add this to the installation instructions for an existing arch install. Thank you.

----STRAP.SH ERRORS----
The first problem with the existing arch install instructions, is that right before running the command "sudo ./strap.sh", you must run the command "chmod +x strap.sh". This is a very easy fix, it just should be noted on the installation page

(Highly recommended to use "sudo blackman -a" rather than "sudo pacman -S blackarch"
----INSTALLING BLACKARCH PACKAGE ERRORS---- (a workaround is just to use blackman)
All existing arch users MUST add the following to their /etc/pacman.conf else they will not be able to "sudo pacman -S blackarch"

Copy+Paste this EXACTLY into your pacman.conf......

[options]
IgnorePkg = base64dump wireshark-cli tcpdump scapy ruby-record tor ruby-uri-query_params sb0x sipsak spectools speedpwn splint sshatter sshscan sslcat ssldump sslh sslsniff steghide swaks truecrypt tcpick tcpjunk tcpreplay timegen tinc tnscmd10g torsocks traceroute treasure trid-defs trid unhide unicornscan upx vanguard vidalia vnc-bypauth voipong vulscan w3af wi-feye wireshark-gtk wmat xf86-video-qxl-git xplico gitrob bettercap

After copy+pasting that, run "sudo pacman -Syy",
Then, when running "sudo pacman -S blackarch", you must spam the "n" key along with "enter". Be careful not to accidentally press "n" when it finally asks if you want to install everything else.

This issue happens because many packages in the blackarch repository have bad mirrors, or they exceed the maximum size for packages, so adding them to your IgnorePkg will attempt to ignore them when installing everything else, thus fixing errors galore. The trick is that once you got the main blackarch install finished, you can just manually install all the packages you ignored (except gitrob and bettercap, because they have extreme incompatabilities with ruby)

This should fix all errors. If you run into anything else problematic during your installation, please post here and I will get back to you soon.

**Comment by CommanderTurtle** [comment]
I should note that the IgnorePkg in pacman.conf is only necessary when running a 32-bit arch installation. Should 64 bit come across any errors, they are few and far between, thus it is easy to download those packages manually

**Comment by auscompgeek** [comment]
> wireshark-cli tcpdump scapy splint sslh steghide swaks truecrypt tcpreplay sslh tinc torsocks mtr traceroute unhide upx vulscan wireshark-gtk

These packages are in the Arch repos. If you have trouble downloading these packages, you need to fix your mirrorlist.

**Comment by yangming** [comment]
new error:
sudo sh strap.sh
[+] installing blackarch keyring...
[-] ERROR: invalid keyring signature. please stop by irc.freenode.net/blackarch

check the line manually:
gpg \                                                                  57 ↵
        --keyserver-options no-auto-key-retrieve \
        --with-fingerprint blackarch-keyring.pkg.tar.xz.sig
gpg: assuming signed data in 'blackarch-keyring.pkg.tar.xz'
gpg: Signature made Mon 20 Jan 2014 12:36:55 AM CST using RSA key ID EA87E4E3
gpg: Good signature from "Evan Teitelman <teitelmanevan@gmail.com>" [unknown]
gpg: WARNING: This key is not certified with a trusted signature!
gpg:          There is no indication that the signature belongs to the owner.
Primary key fingerprint: 4345 7715 66D7 6038 C7FE  B438 63EC 0ADB EA87 E4E3

any idea?

**Comment by rbellamy** [comment]
@yangming my guess is that the line before that is failing.

Try running `gpg --keyserver hkp://pgp.mit.edu --recv-keys 4345771566D76038C7FEB43863EC0ADBEA87E4E3` first, and see if it works.

If you get an error like: `gpg: keyblock resource '/root/.config/gnupg/pubring.kbx': No such file or directory` this means you need to create the `gnupg` config directory.

**Comment by dextr0se** [comment]
Solved the problem with strap.sh
It is a typo in the script

sudo sh strap.sh
[+] installing blackarch keyring...
[-] ERROR: invalid keyring signature. please stop by irc.freenode.net/blackarch

change the the line in the strap.sh file that points to **hkp:**//pgp.mit.edu to **http**://

**Comment by auscompgeek** [comment]
@dextr0se that's not a typo

**Comment by dextr0se** [comment]
@auscompgeek you are correct. 
When I change from the hkp protocol to http to the MIT key server the script worked.  As far as I know the newer hkp servers are running over the http protocol and they are suppose the be backwards compatible with hkp.  It may be a firewall configuration dropping the packets because of a high port to high port.  Anyways, it works when I changed the hkp to http.

**Comment by DistantParadox** [comment]
When I run the strap.sh file after running a chmod +x on it, it just hangs there and doesn't finish:

```
$ sudo ./strap.sh
[+] installing blackarch keyring...

```

I'm following the instructions in "Installing on top of ArchLinux" [here](https://www.blackarch.org/downloads.html#install-repo).  I have also modified my pacman.conf as msentioned above.  Does anybody know how to get past this?

**Comment by DistantParadox** [comment]
I had the same issue as previously mentioned, yet when I run

`gpg --keyserver hkp://pgp.mit.edu --recv-keys 4345771566D76038C7FEB43863EC0ADBEA87E4E3`

I get 

`gpg: keyserver receive failed: Connection timed out`

Might this be a problem on my end?

**Comment by ldionmarcil** [comment]
Please don't bump old (CLOSED!) threads. For stuff like connectivity issue, please check out our IRC channel for help.

**Comment by joelthepirate** [comment]
OMFG Thank you for giving me the idea... I have been wanting to hoard all that BlackArch has to offer for years already and never had success. From your idea and lots of trial and error, I worked out a more updated list of things you must disable (and just a heads up... I love BetterCap... there were no Ruby problems as was previously suggested, but rather... there were conflicting packages). So, I removed the ones that seemed less interesting and left the ones that seemed cooler. Here... add this to your pacman.conf and everything should be golden:

[options]
IgnorePkg = sslyze wafp zackattack androick armitage armscgen autopwn birp bokken chopshop dutas flowinspect gerix-wifi-cracker mitmf nsia pythem smbrelay gitrob ssrf-proxy bvi python2-flask-pymongo mausezahn ncpfs hungry-interceptor unicorn-powershell

Then:
pacman -Syy
ENTER
pacman -S blackarch
ENTER
Answer 'n' to these 23 packages you opted to ignore (via pacman.conf):
[root@penis etc]# pacman -S blackarch
:: androick is in IgnorePkg/IgnoreGroup. Install anyway? [Y/n] n
:: armitage is in IgnorePkg/IgnoreGroup. Install anyway? [Y/n] n
:: armscgen is in IgnorePkg/IgnoreGroup. Install anyway? [Y/n] n
:: autopwn is in IgnorePkg/IgnoreGroup. Install anyway? [Y/n] n
:: birp is in IgnorePkg/IgnoreGroup. Install anyway? [Y/n] n
:: bokken is in IgnorePkg/IgnoreGroup. Install anyway? [Y/n] n
:: bvi is in IgnorePkg/IgnoreGroup. Install anyway? [Y/n] n
:: chopshop is in IgnorePkg/IgnoreGroup. Install anyway? [Y/n] n
:: dutas is in IgnorePkg/IgnoreGroup. Install anyway? [Y/n] n
:: flowinspect is in IgnorePkg/IgnoreGroup. Install anyway? [Y/n] n
:: gerix-wifi-cracker is in IgnorePkg/IgnoreGroup. Install anyway? [Y/n] n
:: gitrob is in IgnorePkg/IgnoreGroup. Install anyway? [Y/n] n
:: hungry-interceptor is in IgnorePkg/IgnoreGroup. Install anyway? [Y/n] n
:: mausezahn is in IgnorePkg/IgnoreGroup. Install anyway? [Y/n] n
:: mitmf is in IgnorePkg/IgnoreGroup. Install anyway? [Y/n] n
:: ncpfs is in IgnorePkg/IgnoreGroup. Install anyway? [Y/n] n
:: nsia is in IgnorePkg/IgnoreGroup. Install anyway? [Y/n] n
:: pythem is in IgnorePkg/IgnoreGroup. Install anyway? [Y/n] n
:: smbrelay is in IgnorePkg/IgnoreGroup. Install anyway? [Y/n] n
:: sslyze is in IgnorePkg/IgnoreGroup. Install anyway? [Y/n] n
:: ssrf-proxy is in IgnorePkg/IgnoreGroup. Install anyway? [Y/n] n
:: unicorn-powershell is in IgnorePkg/IgnoreGroup. Install anyway? [Y/n] n
:: wafp is in IgnorePkg/IgnoreGroup. Install anyway? [Y/n] n
:: zackattack is in IgnorePkg/IgnoreGroup. Install anyway? [Y/n] n

Accept all defaults, BOOM... Pwned!!!
